backend/app/adapter/db/url.go

```diff
@@ -78,6 +79,47 @@ WHERE "%s"=$1;`,
 	return url, nil
 }
 
+// GetURLByVisibility implements retrieval of URL associated with userEmail based on type specified
+// TODO: Add urlType based querying after column addition
+func (u URLSql) GetURLByVisibility(userEmail, urlType string) ([]search.URLSearchResponse, error) {
+	statement := fmt.Sprintf(`
+	SELECT *
+	FROM "%s" 
+	WHERE "%s"
+	IN (SELECT "%s" FROM "%s" WHERE "%s"=$1);`,
+		table.URL.TableName,
+		table.URL.ColumnAlias,
+		table.UserURLRelation.ColumnURLAlias,
+		table.UserURLRelation.TableName,
+		table.UserURLRelation.ColumnUserEmail,
+	)
+
+	var searchResponse []search.URLSearchResponse
```

Comment:
Let's decouple search related logic from this function following single responsibility principle.

Reply:
Okay will create other function in this file to handle search on DB and this function will parse the response and return to the caller, Will that be fine?

Reply:
@lucifercr07 Search is business logic, which should be put inside use case. Let add add search related logic in a separate PR.